src/gui/src/components/navigation/sidebar.tsx

```diff
@@ -0,0 +1,785 @@
+import { useEffect, useState, type MouseEvent } from 'react'
+import { useTheme } from '@/components/ui/theme-provider.jsx'
+import {
+  type LucideIcon,
+  LayoutDashboard,
+  ArrowRightFromLine,
+  ArrowLeftFromLine,
+  ChevronRight,
+  Folder,
+  FolderOpen,
+  Library,
+  Sun,
+  Moon,
+  Menu,
+  X as IconX,
+} from 'lucide-react'
+import { useAnimate, AnimatePresence, motion } from 'framer-motion'
+import { DEFAULT_QUERY, useGraphStore } from '@/state/index.js'
+import {
+  type DashboardDataProject,
+  type Action,
+} from '@/state/types.js'
+
+type selectProjectItem = {
+  updateActiveRoute: Action['updateActiveRoute']
+  updateErrorCause: Action['updateErrorCause']
+  updateQuery: Action['updateQuery']
+  updateStamp: Action['updateStamp']
+  item: DashboardDataProject
+}
+
+interface SidebarLink {
+  name: string
+  href: string
+  icon: LucideIcon
+  target?: 'blank'
+  external: boolean
+}
+
+interface SidebarScreenProps {
+  isOpen: boolean
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>
+  animate: boolean
+  setAnimate: (locked: boolean) => void
+}
+
+const sidebarLinks: SidebarLink[] = [
+  {
+    name: 'Dashboard',
+    href: '/dashboard',
+    icon: LayoutDashboard,
+    external: false,
+  },
+  {
+    name: 'Docs',
+    href: 'https://docs.vlt.sh/cli/commands/query',
+    icon: Library,
+    target: 'blank',
+    external: true,
+  },
+]
+
+/**
+ * Constants placed here for brevity
+ */
+const SIDEBAR_WIDTH = {
+  open: '200px',
+  closed: '60px',
+}
+
+/**
+ * The `animate` prop controls whether the sidebar should either:
+ * be in an open state, or be in a closed state, without triggering
+ * animations.
+ *
+ * Two different sidebars are rendered on < `sm` and > `md` breakpoints.
+ */
+const Sidebar = () => {
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const { lockSidebar: animate, updateLockSidebar: setAnimate } =
+    useGraphStore()
+
+  return (
+    <>
+      <Sidebar.Desktop
+        isOpen={isOpen}
+        setIsOpen={setIsOpen}
+        animate={animate}
+        setAnimate={setAnimate}
+      />
+      <Sidebar.Mobile isOpen={isOpen} setIsOpen={setIsOpen} />
+    </>
+  )
+}
+
+Sidebar.Desktop = ({
+  isOpen,
+  setIsOpen,
+  animate,
+  setAnimate,
+}: SidebarScreenProps) => {
+  const {
+    updateActiveRoute,
+    updateErrorCause,
+    updateQuery,
+    updateStamp,
+    savedProjects,
+  } = useGraphStore()
```

Comment:
Same as elsewhere, let's use the current standard for retrieving properties from the store.